apps/src/aichat/aichatApi.ts

```diff
@@ -101,3 +102,33 @@ export async function postAichatCheckSafety(
 
   return await response.json();
 }
+
+/**
+ * This function sends a GET request to the aichat student chat history backend controller, then returns
+ * a list of chat events (JSON) if successful.
+ */
+export async function getStudentChatHistory(
+  studentUserId: number,
+  levelId: number,
+  scriptId: number | null,
+  scriptLevelId: number | undefined
+): Promise<string[]> {
+  let params: Record<string, string> = {
+    studentUserId: studentUserId.toString(),
+    levelId: levelId.toString(),
+    scriptId: scriptId?.toString() || '',
+  };
+  if (scriptLevelId) {
+    params = {...params, scriptLevelId: scriptLevelId.toString()};
+  }
```

Comment:
this is fine, but could also just be:
```
if (scriptLevelid) {
  params.scriptLevelId = scriptLevelId.toString();
}
```

that way `params` could be declared as `const` on line 116 and we wouldn't be copying and creating a new object (which is super cheap and all, but still)

Reply:
Nice - will update.